ledger/voters.go

```diff
@@ -186,14 +179,39 @@ func (vt *votersTracker) newBlock(hdr bookkeeping.BlockHeader) {
 	}
 }
 
-// lowestRound() returns the lowest round state (blocks and accounts) needed by
+// removeOldVoters removes voters data form the tracker and allows the database to commit previous rounds.
+// voters would be removed if one of the two condition is met
+// 1 - Voters are for a round which was already been confirmed by stateproof
+// 2 - Voters are for a round which is older than the allowed recovery interval.
+// notice that if state proof chain is delayed, votersForRoundCache will not be larger than
+// StateProofRecoveryInterval + 1
+// ( In order to be able to build and verify X stateproofs back we need X + 1 voters data )
+func (vt *votersTracker) removeOldVoters(hdr bookkeeping.BlockHeader) {
+	// we calculate the lowest round for recovery according to the newest round (might be different from the rounds on cache)
+	proto := config.Consensus[hdr.CurrentProtocol]
+	recentRoundOnRecoveryPeriod := basics.Round(uint64(hdr.Round) - uint64(hdr.Round)%proto.StateProofInterval)
+	oldestRoundOnRecoveryPeriod := recentRoundOnRecoveryPeriod.SubSaturate(basics.Round(proto.StateProofInterval * proto.StateProofRecoveryInterval))
+
+	for r, tr := range vt.votersForRoundCache {
```

Comment:
Basically, You are right, and the logic you've suggested will work fine.
However, I think that the current approach is simpler (we don't need a store state for example- and we might have more edge cases).
Since the map is quite small (it usually contains 0-3 elements and in the worst case contains 10 elements). I suggest we stick to a simpler code.